### ISSUE TITLE
Update dartdoc documentation to reflect that pub get is required.

### ISSUE THREAD

**Issue by bsutton** [dart-lang/dartdoc#2992]
I've just spent close to a day trying to analyze a problem with dartdoc failing with the following error:

```
dartdoc --no-validate-links  --output $pathToDocRoot --input $pathToExpandedTar

Failed to generate doc for packageStore[/public/bo/xbxpsdavuh/bonsoir/1.0.1]()+2.tar.gz.
Error Details:
Documenting bonsoir...
error: unresolved export uri: "package:bonsoir[/src/broadcast.dart]()"
from bonsoir:
../…/lib/bonsoir.dart:1
Initialized dartdoc with 39 libraries

```
I eventually worked out that you MUST run `pub get` before running dart doc.
I've re-checked the documentation but it makes no mention of this fact.

Can we change the doco to mention the requirement and have dartdoc throw an meaningful error of pub get hasn't been run.

To reproduce this problem.

git clone a dart project (or unpack a tar).
run dartdoc.
Errors ensure.

Dart version 2.16.1
Running dart doc from `pub global activate dartdoc` version 5.0.1.
Linux 64 bit system.

**Comment by parlough** [comment]
Ah, not running `pub get` automatically was introduced in the 5.0.0 release of `dartdoc`: https://github.com/dart-lang/dartdoc/blob/master/CHANGELOG.md#500

I'll add something to dart.dev but a meaningful error would be great to.

\cc @devoncarew Since I think you made this change.

**Comment by bsutton** [comment]
OK, thanks.

I have bruises on my forehead over this one :)

**Comment by devoncarew** [comment]
Thanks for the feedback! I agree we should update the docs in various to reflect this change.

Note that other tools (dart fix, dart analyze, ...) also don't run pub when they're executed. They do give good failure messages in those cases though (xxx analysis issues found), so it's easier to diagnose what's going wrong.

 We should likely expose the number of analysis issues - or something similar - in dartdoc as part of a run.

**Comment by srawlins** [comment]
Fixed with #2996
